### ISSUE TITLE
[TECHNICAL] Changes in accessibility role management

### ISSUE THREAD

**Issue by JuancaG05** [owncloud/android#4462]
Regarding this PR https://github.com/owncloud/android/pull/4454, there are several things that can be improved:

- Include `contentDescription` as another parameter in the `setAccessibilityRole` extension function, so that we avoid implementing `AccessibilityDelegateCompat` again, like in `SortOptionsView`
- Check if `onCreateOptionsMenu` can actually be moved to `FileFragment` and remove it from its subclasses
- When setting `contentDescription` as a concatenation of several Strings, add a space in between so that they form a sentence correctly. Make it in a more Kotlinized way
- String `<string name="content_description_add_new_content_expanded">"Add new content expanded"</string>` shouldn't have the `"`
- String `<string name="content_description_sort_by_name_descending">sort by %1$s descending</string>` should start by a capital letter to be consistent with its analogue `content_description_sort_by_name_ascending`

Other refactor that has to do with accessibility but not with role management, regarding https://github.com/owncloud/android/pull/4455:
- `KeyEvent.KEYCODE_TAB` shouldn't be handled like that, since it breaks the navigation model. `Tab` should be used to move to the next element in the view, not moving to `lock_time` (which is not always visible)
- `KeyEvent.KEYCODE_DPAD_DOWN` should just move the focus from the toolbar to the first element in the view, if possible, not moving it straight to the custom number keyboard shown in the view. That's how the navigation model works in other views
- Check why introducing the numbers with the keyboard makes the passcode visible

### TASKS

- [x] Research (if needed)
- [x] Create branch technical/refactor_accessibility_role
- [x] Development tasks
  - [x] Include `contentDescription` as parameter in `setAccessibilityRole`
  - [x] Check `onCreateOptionsMenu` in `FileFragment` feasibility
  - [x] Kotlinize Strings concatentation and add missing spaces in between
  - [x] Fix `content_description_add_new_content_expanded` String
  - [x] Fix `content_description_sort_by_name_descending` String
  - [x] Check `KeyEvent.KEYCODE_TAB` handling
  - [x] Check `KeyEvent.KEYCODE_DPAD_DOWN` handling
  - [x] Check passcode visibility when introducing numbers from the keyboard
  - [x] Implement unit tests (if needed)
- [x] Code review and apply changes requested
- [x] Design test plan
- [x] QA
- [ ] Merge branch technical/refactor_accessibility_role into master
